### ISSUE TITLE
CentavosInputFormatter - Possibilidade de informar casas decimais diferente de 2 ou 3

### ISSUE THREAD

**Issue by altevircardosoneto** [flutterbootcamp/brasil_fields#91]
Primeiramente parabéns pelo package, tem ajudado muito nos projetos! 🚀🚀🚀
É possível fazer com que o formatter CentavosInputFormatter receba casas decimais dinamicamente, além de 2 ou 3 casas decimais como é hoje?
Não é comum, porém, peguei alguns cenários/projetos onde a quantidade de casas decimais é definido na API, com base nesse dado retornado para o aplicativo o campo deveria ser formatado com 4 casas decimais por exemplo.

**Comment by rubensdemelo** [comment]
Ola @Altevir , obrigado.

Veja se o IOFInputFormatter te atende: https://pub.dev/documentation/brasil_fields/latest/brasil_fields/IOFInputFormatter-class.html

**Comment by altevircardosoneto** [comment]
@rubensdemelo 
Infelizmente não atende, esse formatter só gera 1 casa antes da vírgula, teria que ser no formato do CentavosInputFormatter mesmo.

**Comment by rubensdemelo** [comment]
Mas qual seria o intuito? poder formatar um valor monetário com mais de 3 casas decimais ? 

R$ 3.343.221,892

**Comment by altevircardosoneto** [comment]
@rubensdemelo 
Exatamente, tendo a possibilidade de ser dinâmico o número de casas decimais e não fixo.

**Comment by rubensdemelo** [comment]
@Altevir Mas nao tem como ser dinâmico e formatar nas posições corretas.

Como formatar dinamicamente o valor 3434392 ?

34.343,92
3.434.392
3.434,392

Por isso o parametro de casas decimais é um parametro explicito.

**Comment by altevircardosoneto** [comment]
@rubensdemelo 
Na realidade tem, até fiz isso quando vi que o package não tinha essa possibilidade, criei uma classe chamada CurrencyInputFormatter (baseada no CentavosInputFormatter) tendo como parâmetro o número de casas decimais e informando a esse inputFormatter do TextFormField via ValueNotifier, funcionou 100%.

Exemplo:
final ValueNotifier<int> quantCasasDecimais = ValueNotifier<int>(2);

quantCasasDecimais pode vir de uma configuração ou regra da API.

TextFormField(
    textInputAction: TextInputAction.done,
    keyboardType: TextInputType.number,
    inputFormatters: [
      FilteringTextInputFormatter.digitsOnly,
      CurrencyInputFormatter(casasDecimais: quantCasasDecimais.value),
    ],
)

**Comment by rubensdemelo** [comment]
Entao, vc esta certo. Por isso falei que a quantidade de casas decimais precisa ser um parametro explicito. E este parametro (pode) ser dinamico.

**Comment by altevircardosoneto** [comment]
@rubensdemelo 
Exato, só quis exemplificar pra explicar melhor o conceito.
